src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -307,158 +379,101 @@ public override IAsyncResult BeginGetResponse(AsyncCallback callback, object sta
         /// <returns>A System.Net.WebResponse that contains the response from the Internet resource.</returns>

```

Comment:
I think the docstring needs to be updated. `ConvertHttpClientResponse` returns `HttpWebResponseMessage`